tests/security/tpm2/tpm2_env_setup.pm

```diff
@@ -18,19 +18,30 @@ sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
-    # Add user tss, tss is the default user to start tpm2.0 service
-    my $user = "tss";
-    my $passwd = "susetesting";
-    assert_script_run "useradd -d /home/$user -m $user";
-    assert_script_run "echo $user:$passwd | chpasswd";
-
     # Install the tpm2.0 related packages
     # and then start the TPM2 Access Broker & Resource Manager
-    zypper_call "in expect ibmswtpm2 tpm2.0-abrmd tpm2.0-abrmd-devel openssl tpm2-0-tss tpm2-tss-engine tpm2.0-tools";
+    zypper_call("in expect ibmswtpm2 tpm2.0-abrmd tpm2.0-abrmd-devel openssl tpm2-0-tss tpm2-tss-engine tpm2.0-tools");
+
+    # Add user tss, tss is the default user to start tpm2.0 service
+    # However, a daemon user may be created as well during tss related
+    # packages installation, we can re-use it then with some changes.
+
+    my $tss_user = "tss";
+    if (script_run("id $tss_user") == 0) {
+        my $tss_home = script_output("cat /etc/passwd | grep $tss_user | cut -d : -f6");
+        assert_script_run("mkdir -p $tss_home");
+        assert_script_run("usermod -s /bin/bash $tss_user");
+        assert_script_run("chown $tss_user $tss_home");
+    }
+    else {
+        assert_script_run("mkdir -p /home/$tss_user");
```

Comment:
`useradd` is already called with `-m`, so no mkdir necessary

Reply:
Good suggestion, let me enhance my code